terraform/provider/resource_teleport_app.go

```diff
@@ -102,7 +103,21 @@ func (r resourceTeleportApp) Create(ctx context.Context, req tfsdk.CreateResourc
 	}
 
 	id := app.Metadata.Name
-	appI, err := r.p.Client.GetApp(ctx, id)
+	var appI apitypes.Application
+
+	backoff := backoff.NewDecorr(r.p.RetryConfig.Base, r.p.RetryConfig.Cap, clockwork.NewRealClock())
```

Comment:
What is the maximum retry duration? I.e. after how long will we give up retrying?

Reply:
I didn't set any limit here
This delegates the actual maximum to the one coming from Terraform (trough the `context.Done()`

The sdkv2 has a timeout of 20 minutes
https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts#default-timeouts-and-deadline-exceeded-errors

However, we are using plugin framework and not sdkv2, which seems to have a higher timeout
![image](https://user-images.githubusercontent.com/689271/168248810-25abc1d9-7942-483e-a9f9-e642ef914a1c.png)

I guess we have two options here:
- let it run anyway and wait for the parent executor (whoever calls the `terraform apply`) to cancel the request
- have an explicit configurable timeout or max number of retries

I'll add a max number of retries